src/server/builder.ts

```diff
@@ -223,7 +223,7 @@ namespace ts.server {
             for (const reference of this.references) {
                 reference.removeReferencedBy(this);
             }
-            this.references = createSortedArray<ModuleBuilderFileInfo>();
+            clear(this.references);

```

Comment:
So these two lines are meaningfully different in behavior - previously anything that saved a reference to `this.references` prior to reassignment kept the old array, but now we mutate rather than reassigning. Is that OK to do here?